flow/scenarios/base_scenario.py

```diff
@@ -58,6 +58,243 @@ class Scenario(Serializable):
+        network. For more, see flow/core/params.py
+    nodes : list of dict or None
+        list of nodes that are assigned to the scenario via the `specify_nodes`
+        method. All nodes in this variable are expected to have the following
+        properties:
+
+        * **name**: a unique identifier for the node
+        * **x**: x-coordinate of the node, in meters
+        * **y**: y-coordinate of the node, in meters
+
+        If the scenario is meant to generate the network from an OpenStreetMap
+        or template file, this variable is set to None
+    edges : list of dict or None
+        edges that are assigned to the scenario via the `specify_edges` method.
+        This include the shape, position, and properties of all edges in the
+        network. These properties include the following mandatory properties:
+
+        * **id**: name of the edge
+        * **from**: name of the node the edge starts from
+        * **to**: the name of the node the edges ends at
+        * **length**: length of the edge
+
+        In addition, either the following properties need to be specifically
+        defined or a **type** variable property must be defined with equivalent
+        attributes in `self.types`:
+
+        * **numLanes**: the number of lanes on the edge
+        * **speed**: the speed limit for vehicles on the edge
+
+        Moreover, the following attributes may optionally be available:
+
+        * **shape**: the positions of intermediary nodes used to define the
+          shape of an edge. If no shape is specified, then the edge will appear
+          as a straight line.
+
+        Note that, if the scenario is meant to generate the network from an
+        OpenStreetMap or template file, this variable is set to None
+    types : list of dict or None
+        A variable used to ease the definition of the properties of various
+        edges. Each element in the list consists of a dict consisting of the
+        following property:
+
+        * **id**: name of the edge type. Edges in the `self.edges` attribute
+          with a similar value under the "type" key will adopt the properties
+          of other components of this list, such as "speed" and "numLanes".
+
+        If the type variable is None, then no types are available within the
+        scenario.
+
+        Note that, if the scenario is meant to generate the network from an
+        OpenStreetMap or template file, this variable is set to None
+    connections : list of dict or None
+        TODO
+
+        If the connections attribute is set to None, then the connections
+        within the network will be specified by the simulator.
+
+        Note that, if the scenario is meant to generate the network from an
+        OpenStreetMap or template file, this variable is set to None
+    routes : dict
+        A variable whose keys are the starting edge of a specific route and the
+        element is the list of edges a vehicle is meant to traverse starting
```

Comment:
Replace with "A variable whose keys are the starting edge of a specific route, and whose values are...." 'Values' is the proper Python terminology for dictionaries.

Reply:
done